api/v1beta1/common_types.go

```diff
@@ -173,6 +183,33 @@ const (
 	VariableDefinitionFromInline = "inline"
 )
 
+// MachineSetPreflightCheck defines a valid MachineSet preflight check.
+type MachineSetPreflightCheck string
+
+const (
+	// MachineSetPreflightCheckAll can be used to represent all the MachineSet preflight checks.
+	MachineSetPreflightCheckAll MachineSetPreflightCheck = "All"
+
+	// MachineSetPreflightCheckKubeadmVersion is the name of the kubeadm-version preflight check.
+	// It checks that the MachineSet and the ControlPlane conform to the kubeadm version skew policy.
+	// The preflight check is only run if a ControlPlane is used (controlPlaneRef must exist in the Cluster),
+	// the ControlPlane has a version, the MachineSet has a version and the MachineSet uses the Kubeadm bootstrap
+	// provider.
+	MachineSetPreflightCheckKubeadmVersion MachineSetPreflightCheck = "KubeadmVersion"
```

Comment:
I would have aligned to kubeadm which has a precheck called "KubernetesVersion". But not strong opinion if you prefer "KubernetesVersionSkew" and "KubeadmVersionSkew"

Reply:
I think we should look at this from the user's perspective, and make it clear what kind of validation this check is performing; consistency with Kubeadm is less of a concern IMO, mostly because there is nothing enforcing consistency between the implementation of the checks (they might diverge over time, if they are not already).

In this case, it is validating we are consistent with the KubeadmVersion Skew policy, so I think adding skew is a valuable addition.